tests/utils.py

```diff
@@ -125,34 +132,99 @@ def clean_layout(layout):
+                #   '\n'.format(n=layout)
+                # + '  ' * depth
+                # + '  border: {n[border_box][size][0]}x{n[border_box][size][1]}'
+                #   ' @ ({n[border_box][position][0]}, {n[border_box][position][1]})'
+                #   '\n'.format(n=layout)
+                + ''.join(
+                        output_layout(child, depth=depth + 1)
+                        for child in layout.get('children', [])
+                    ) if layout else ''
+                + ('\n' if layout and layout.get('children', None) and depth > 1 else ''))
     else:
-        return ('  ' * depth
-            + "* '{text}'\n".format(text=layout['text'].strip())
-        )
+        return ('  ' * depth + "* '{text}'\n".format(text=layout['text'].strip()))
+
+
+def copy_fonts(system=False):
+    """Copy needed files for running tests."""
+    fonts_folder = FontDatabase.fonts_path(system=system)
+
+    if not os.path.isdir(fonts_folder):
+        os.makedirs(fonts_folder)
+
+    fonts_data_path = os.path.join(HERE, 'data', 'fonts')
+    font_files = sorted([item for item in os.listdir(fonts_data_path) if item.endswith('.ttf')])
+    for font_file in font_files:
+        font_file_data_path = os.path.join(fonts_data_path, font_file)
+        font_file_path = os.path.join(fonts_folder, font_file)
+
+        if not os.path.isfile(font_file_path):
+            shutil.copyfile(font_file_data_path, font_file_path)
+        # Register font
+
+        if os.name == 'nt':
+            import winreg  # noqa
+            base_key = winreg.HKEY_LOCAL_MACHINE if system else winreg.HKEY_CURRENT_USER
+            key_path = r"Software\Microsoft\Windows NT\CurrentVersion\Fonts"
+
+            if '_' in font_file:
+                font_name = font_file.split('_')[-1].split('.')[0]
+            else:
+                font_name = font_file.split('.')[0]
 
+            # This font has a space in its system name
+            if font_name == 'WhiteSpace':
+                font_name = 'White Space'
 
-class LayoutTestCase(TestCase):
+            font_name = font_name + ' (TrueType)'
+
+            with winreg.OpenKey(base_key, key_path, 0, winreg.KEY_ALL_ACCESS) as reg_key:
+                value = None
+                try:
+                    # Query if it exists
+                    value = winreg.QueryValueEx(reg_key, font_name)
+                except FileNotFoundError:
+                    pass
+
+                # If it does not exists, add value
+                if value != font_file_path:
+                    winreg.SetValueEx(reg_key, font_name, 0, winreg.REG_SZ, font_file_path)
+
+
+class ColosseumTestCase(TestCase):
+    """Install test fonts before running tests that use them."""
+    _FONTS_ACTIVE = False
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self._FONTS_ACTIVE is False:
+            self.copy_fonts()
+
+    def copy_fonts(self):
+        copy_fonts()
```

Comment:
Any reason for this "copy_fonts calls copy fonts" indirection?

Reply:
I am using copy_fonts (the func) also on the `if __init__ ...` (for CI)